applicationset/generators/git_test.go

```diff
@@ -1466,21 +1465,93 @@ func TestGitGenerator_GenerateParams(t *testing.T) {
 			expected:           []map[string]any{{"path": "app1", "path.basename": "app1", "path.basenameNormalized": "app1", "path[0]": "app1", "values.foo": "bar"}},
 			expectedError:      errors.New("error getting project project: appprojects.argoproj.io \"project\" not found"),
 		},
+		{
+			name: "Project field is not templated - verify that project is passed through to repo-server as-is",
```

Comment:
I am not sure I understand the tests. The new appProject params does not seem to be used.
The two tests dont seem to assert on whether the project is passed when non-templated or is "" when templating. The both call a mock that mocks `mock.Anything` that will always return the same result.